controller/control.go

```diff
@@ -134,6 +134,16 @@ func (c *Controller) UpdateVolStatus() {
 	} else {
 		c.ReadOnly = true
 	}
+	if rwReplicaCount == c.ReplicationFactor {
+		checkpoint, err := c.backend.GetLastSnapshot()
+		if err == nil {
```

Comment:
what if 'err' != nil? need to reset Checkpoint as it is got reset if rwReplicaCount is not equal to c.ReplicationFacotr?